src/Components/Search/SearchBar.tsx

```diff
@@ -315,25 +320,67 @@ export class SearchBar extends Component<Props, State> {
     const edges = get(viewer, v => v.search.edges, [])
     const suggestions = xs ? edges : [firstSuggestionPlaceholder, ...edges]
     return (
-      <AutosuggestContainer ref={ref => (this.containerRef = ref)}>
-        <Autosuggest
-          suggestions={suggestions}
-          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
-          onSuggestionHighlighted={this.throttledOnSuggestionHighlighted}
-          onSuggestionsFetchRequested={this.throttledFetch}
-          getSuggestionValue={this.getSuggestionValue}
-          renderSuggestion={this.renderSuggestion}
-          renderSuggestionsContainer={props => {
-            return this.renderSuggestionsContainer(props, { xs })
-          }}
-          inputProps={inputProps}
-          onSuggestionSelected={(e, selection) => {
-            e.preventDefault()
-            this.onSuggestionSelected(selection)
-          }}
-          renderInputComponent={this.renderInputComponent}
-        />
-      </AutosuggestContainer>
+      <Subscribe to={[SearchBarState]}>
+        {(searchState: SearchBarState) => {
+          const { state } = searchState
+          return (
+            <AutosuggestContainer
+              onKeyDown={e => {
```

Comment:
Would probably be a little more tidy to move this into a `handleKeyboardNavigation` method; or better -- rename  `AutosuggestContainer` to `AutosuggestManager` and move it into its own file, which hides away this keyboard handler logic as well as state manipulation. Then can do 

```jsx
<AutosuggestManager>
  <Autosuggest ... />
</AutosuggestManager>
```
and keep the tree clean.

Reply:
Interesting! I like that. Will experiment with some better structure here.

Reply:
+1 it'd be nice to extract this somewhere.